File: src-tauri/src/core/onboarding.rs
use crate::{care_instructions, config::get_config, medications, pro, resources};
use rusqlite::Connection;
use std::fs;
use tauri::{AppHandle, Manager};

use super::schema;

fn combine_schemas() -> String {
    let combined_schemas = schema::CORE_SCHEMA.to_string() + care_instructions::schema::CARE_INSTRUCTIONS_SCHEMA + medications::schema::MEDICATIONS_SCHEMA + pro::schema::PRO_SCHEMA + resources::schema::RESOURCES_SCHEMA;
    combined_schemas
}

pub fn setup_onboarding(app: &AppHandle) {
    let app_data_dir = app.path().app_data_dir().unwrap();

    if !app_data_dir.join("iris.db").exists() {
        fs::create_dir_all(app_data_dir.join("recordings/")).unwrap();
        get_config(&app);

        let conn = Connection::open(app_data_dir.join("iris.db")).unwrap();

        conn.execute_batch(&combine_schemas())
            .unwrap();
    }
}

File: src/utils/onboarding.ts
import { mkdir, exists } from "@tauri-apps/plugin-fs";
import { userDataDir } from "./folders";
import { join } from "@tauri-apps/api/path";
import { getConfig } from "./settings";
import { connectUserDB } from "./database";
import { CARE_INSTRUCTIONS_SCHEMA } from "@/types/care-instructions";
import { MEDICATION_SCHEMA } from "@/types/medications";
import { PRO_SCHEMA } from "@/types/pro";
import { RESOURCE_SCHEMA } from "@/types/resources";
import { CONTACT_SCHEMA } from "@/types/contacts";

export async function setupOnboarding() {
  const dataDir = await userDataDir();
  const dbPath = await join(dataDir, "iris.db");
  const recordingsPath = await join(dataDir, "recordings");
  const dbExists = await exists(dbPath);

  if (!dbExists) {
    const combinedSchema =
      CARE_INSTRUCTIONS_SCHEMA +
      MEDICATION_SCHEMA +
      PRO_SCHEMA +
      RESOURCE_SCHEMA +
      CONTACT_SCHEMA;

    await mkdir(recordingsPath, { recursive: true });
    await getConfig();

    const conn = await connectUserDB();
    await conn.execute(combinedSchema);
  }
}
